lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -522,6 +520,32 @@ function dosomething_reportback_form_alter(&$form, &$form_state, $form_id) {
   }
 }
 
+/**
+ * Callback for /rbf/FID page.
+ */
+function dosomething_reportback_permalink_view($args) {
+  $rbid = $args[0];
+  $fid = $_GET['fid'];
+
+  if (variable_get('rogue_collection', FALSE)) {
+    return dosomething_reportback_view_rogue($rbid, $fid);
+  }
+
+  return dosomething_reportback_view_entity(entity_load('reportback', $rbid));
+}
+
+/**
+ * Callback for /rbf/FID page, when Rogue is enabled.
+ */
+function dosomething_reportback_view_rogue($rbid, $fid) {
+    // @TODO: Request signup from Rogue so we can display it on permalink page.
+    // $rogue_signup = dosomething_rogue_client()->get('v2/activity')
```

Comment:
Just to understand, is the issue you can't get a specific signup because this returns a whole bunch of signups for the campaign?